controllers/discordactions.js

```diff
@@ -67,25 +66,19 @@ const createGroupRole = async (req, res) => {
 const getAllGroupRoles = async (req, res) => {
   try {
     const { groups } = await discordRolesModel.getAllGroupRoles();
-    const groupsWithMemberCount = await discordRolesModel.getNumberOfMemberForGroups(groups);
-    const groupCreatorIds = groupsWithMemberCount.reduce((ids, group) => {
-      ids.add(group.createdBy);
-      return ids;
-    }, new Set());
-    const groupCreatorsDetails = await retrieveUsers({ userIds: Array.from(groupCreatorIds) });
-    const groupsWithUserDetails = groupsWithMemberCount.map((group) => {
-      const groupCreator = groupCreatorsDetails[group.createdBy];
-      return {
-        ...group,
-        firstName: groupCreator.first_name,
-        lastName: groupCreator.last_name,
-        image: groupCreator.picture?.url,
-      };
-    });
-
+    const dev = req.query.dev === "true";
+    if (dev) {
+      // Placing the new changes under the feature flag.
+      const discordId = req.userData?.discordId;
```

Comment:
if the discordId is undefined then will this function still continue its execution?

Reply:
Hey Vaibhav, 
The middleware that is applied will make sure that we have the discordId else this function will not be called.

`router.get("/groups", authenticate, checkIsVerifiedDiscord, getAllGroupRoles);`

Reply:
Resolving this in favor of the previous comment. 
I hope the explanation was sufficient